agent/api/task.go

```diff
@@ -674,3 +678,18 @@ func (t *Task) String() string {
 	}
 	return res + "]"
 }
+
+func (task *Task) GetID() (string, error) {
+	parsedARN, err := arn.Parse(task.Arn)
+	if err != nil {
+		return "", errors.Wrapf(err, "task get-id: unable to parse taskARN")
+	}
+
+	resource := parsedARN.Resource
+	resourceSplit := strings.SplitN(resource, arnResourceDelimiter, arnResourceSections)
```

Comment:
SplitN is safer here, because you'll end up with a  known number of sections (or less, if there are not enough delimiters); it leaves any extra unparsed.  Here, it will simplify error handling because you only need to check whether there are enough delimited sections rather than also having to handle too many.